pkgs/applications/networking/browsers/firefox/wrapper.nix

```diff
@@ -111,14 +126,55 @@ let
         cp -R --no-preserve=mode,ownership ${browser}/Applications/${browserName}.app $out/Applications
         rm -f $out${browser.execdir or "/bin"}/${browserName}
       '' + ''
-        if [ ! -x "${browser}${browser.execdir or "/bin"}/${browserName}" ]
+
+        # Link the runtime. The executable itself has to be copied,
```

Comment:
Have you looked into patching firefox? While this solution works, it *is* a little hacky.

Reply:
I have, and I find the code to be surprisingly unreadable for a newcomer. I can spend some more time on that, of course.
If somebody else with more knowledge of the mozilla codebase wants to take a shot at that / point me to the right location in the sources, I am quite happy to do that.

Reply:
Have a look at this: https://github.com/mozilla/gecko/blob/72f5488046d2d4cc8c0470f071db8f504629f95b/browser/components/enterprisepolicies/EnterprisePolicies.js#L362

Reply:
For the autoconfig part, I guess the `readConfigFile` and `openAndEvaluateJSFile` here are responsible: https://github.com/mozilla/gecko/blob/72f5488046d2d4cc8c0470f071db8f504629f95b/extensions/pref/autoconfig/src/nsReadConfig.cpp#L120

Reply:
Yes, but where is XREAppDist actually set?
[here](https://github.com/mozilla/gecko/blob/72f5488046d2d4cc8c0470f071db8f504629f95b/toolkit/xre/nsXREDirProvider.cpp#L462) ?

That in turn references NS_GRE_DIR, which comes from aGREDir passed to nsXREDirProvider::Initialize.

[here](https://github.com/mozilla/gecko/blob/72f5488046d2d4cc8c0470f071db8f504629f95b/toolkit/xre/nsAppRunner.cpp#L3074) perhaps?

Reply:
Does it matter? We can just replace it with

```python
configFile = os.environ.get("SOME_ENV_VAR", "")
```

or similar. No need for `XREAppDist`.

Reply:
Ok, I guess thats a valid thing to do.

Reply:
Ok, I will leave this pr open and, if that works out, open a separate pr which does the same thing using patches and environment variables.
My laptop is busy compiling firefox right now, lets see if its done by tomorrow.

Reply:
Takes 1.5hrs [on hydra](https://hydra.nixos.org/build/90727448#tabs-summary), so at least it is not as ridiculous as chromium.

Reply:
Well actually it doesn't compile at all locally on my machine (even the unmodified firefox-unwrapped).
The error is entirely uninformative:
collect2: error: ld returned 1 exit status
make[3]: *** [/build/firefox-65.0.2/config/rules.mk:712: libxul.so] Error 1
make[3]: Leaving directory '/build/firefox-65.0.2/obj-x86_64-pc-linux-gnu/toolkit/library'

Nothing interesting before or after that.
Do you know a way to get more verbose logging or something?

Reply:
If `ld returned 1 exit status`, there may be a more informative error message earlier in the log.

Maybe you ran out of ram or something?

Reply:
A friend of mine lent me some more capable hardware, it compiles now.

Reply:
Nice! Is it working?

Reply:
I got the policies engine working. AutoConfig appears to be more trouble, because of the lock file + obfuscation value + config file proper thing.
Also its cpp instead of javascript, so debugging is harder.

Reply:
Nice! Yeah AutoConfig does seem a little unnecessary complicated.